src/components/Badge/Badge.tsx

```diff
@@ -29,11 +29,11 @@ export const Badge: FC<BadgeProps> = ({
     return (
         // This `tw-inline-block` and the `tw-float-left` on the first child was added, to remove
         // a 1px space before this component when using without the `icon`-prop
-        <span className="tw-inline-block">
+        <span className="tw-inline-flex tw-flex-initial tw-min-w-0">
             <Container
                 onClick={() => onClick && onClick()}
                 className={merge([
-                    'tw-float-left tw-inline-flex tw-items-center tw-justify-center tw-rounded-full tw-transition-color tw-select-none',
+                    'tw-float-left tw-inline-flex tw-items-center tw-justify-center tw-rounded-full tw-transition-color tw-select-none tw-flex-initial tw-min-w-0',
```

Comment:
is float still required?

Reply:
Ah now I see the issue of the inline-block/float-left, let me revisit this